drivers/okhttp/src/main/java/org/jclouds/http/okhttp/DelegatingSocketFactory.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.http.okhttp;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.Socket;
+
+import javax.net.SocketFactory;
+
+public class DelegatingSocketFactory extends SocketFactory {
```

Comment:
Why is this class needed?

Reply:
Hi 

I was referring below issue to fix this issue. 

https://github.com/square/okhttp/issues/6561#issuecomment-782308646
https://github.com/square/okhttp/issues/6561#issuecomment-782292563

Reply:
My point is that unless I'm missing something this `DelegatingSocketFactory` doesn't add any logic and it's doing nothing. Why do we need it?

Reply:
Hi,

Builder.setSSLSocketFactory is deprecated which force to use Builder.setSocketFactory
but setSocketFactory doesnot accept SSLSocketfactory.getDefault() and throws "java.lang.IllegalArgumentException: socketFactory instanceof SSLSocketFactory" at **runtime**

To overcome the above problem, a delegator class called "DelegatingSocketFactory" created which delegates instance of SSLSocketFactory to SocketFactory

Reply:
Thanks for the explanation!
Can you add this motivation in a comment in the code so anyone reading it can easily get why it is done this way?